pkgs/tools/security/step-ca/default.nix

```diff
@@ -2,29 +2,43 @@
 , lib
 , fetchFromGitHub
 , buildGoModule
+, coreutils
 , pcsclite
 , PCSC
 , pkg-config
+, hsmSupport ? true
 }:
 
 buildGoModule rec {
   pname = "step-ca";
-  version = "0.15.6";
+  version = "0.15.8";
 
   src = fetchFromGitHub {
     owner = "smallstep";
     repo = "certificates";
     rev = "v${version}";
-    sha256 = "0n26692ph4q4cmrqammfazmx1k9p2bydwqc57q4hz5ni6jd31zbz";
+    sha256 = "pDnQlRcUbp63AW9XM9tJ7kpiki21ceQdyEBcGIH6a40=";
   };
 
-  vendorSha256 = "0w0phyqymcg2h2jjasxmkf4ryn4y1bqahcy94rs738cqr5ifyfbg";
+  vendorSha256 = "WDZhXxS7gqgmnZiafnJhw66EbhjPfL9AHvo3OtJI1D4=";
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = lib.optional hsmSupport pkg-config;
 
-  buildInputs =
-    lib.optional stdenv.isLinux (lib.getDev pcsclite)
-    ++ lib.optional stdenv.isDarwin PCSC;
+  buildInputs = lib.optional hsmSupport
+    (lib.optional stdenv.isLinux (lib.getDev pcsclite)
+      ++ lib.optional stdenv.isDarwin PCSC);
```

Comment:
```suggestion
  buildInputs = lib.optional hsmSupport
    (lib.optional stdenv.isLinux pcsclite
      ++ lib.optional stdenv.isDarwin PCSC);
```

Reply:
@SuperSandro2000 This leads to
```bash
builder for '/nix/store/ryg7969vam518gicqlq8n0gbyhxyyzsf-step-ca-0.15.8.drv' failed with exit code 1; last 10 log lines:
  github.com/go-piv/piv-go/piv
  # pkg-config --cflags  -- libpcsclite
  Package libpcsclite was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libpcsclite.pc'
  to the PKG_CONFIG_PATH environment variable
  No package 'libpcsclite' found
  pkg-config: exit status 1
  go.step.sm/cli-utils/command/version
  internal/profile
  net/http/pprof
```
What about `pcsclite.dev` instead? I have no idea why it doesn't work though as many other packages do the same.

Reply:
I'm currently switching to `optionals` instead of `optional` and it seems to be working somehow...

Reply:
Ahh, I got it. This creates a nested list and somehow it worked before.

Reply:
I fixed it. Traded some minimal duplication for easier readability. I hope this is fine.